--- crypto/asn1/a_object.c.orig
+++ crypto/asn1/a_object.c
@@ -428,6 +428,7 @@ i2t_ASN1_OBJECT_cbb(const ASN1_OBJECT *aobj, CBB *cbb
 int
 i2t_ASN1_OBJECT_internal(const ASN1_OBJECT *aobj, char *buf, int buf_len, int no_name)
 {
+	char throwaway;
 	uint8_t *data = NULL;
 	size_t data_len;
 	CBB cbb;
@@ -435,6 +436,13 @@ i2t_ASN1_OBJECT_internal(const ASN1_OBJECT *aobj, char
 
 	if (buf_len < 0)
 		return 0;
+
+	if (buf == NULL) {
+		if (buf_len != 0)
+			return 0;
+		buf = &throwaway;
+	}
+
 	if (buf_len > 0)
 		buf[0] = '\0';
 
--- crypto/bn/bn_add.c.orig
+++ crypto/bn/bn_add.c
@@ -169,6 +169,20 @@ BN_uadd(BIGNUM *r, const BIGNUM *a, const BIGNUM *b)
 		rn = b->top;
 	if (rn == INT_MAX)
 		return 0;
+
+	if (a->top == 0) {
+		if (!bn_copy(r, b))
+			return 0;
+		r->neg = 0;
+		return 1;
+	}
+	if (b->top == 0) {
+		if (!bn_copy(r, a))
+			return 0;
+		r->neg = 0;
+		return 1;
+	}
+
 	if (!bn_wexpand(r, rn + 1))
 		return 0;
 
@@ -192,6 +206,14 @@ BN_usub(BIGNUM *r, const BIGNUM *a, const BIGNUM *b)
 		BNerror(BN_R_ARG2_LT_ARG3);
 		return 0;
 	}
+
+	if (b->top == 0) {
+		if (!bn_copy(r, a))
+			return 0;
+		r->neg = 0;
+		return 1;
+	}
+
 	rn = a->top;
 
 	if (!bn_wexpand(r, rn))
--- crypto/bn/bn_convert.c.orig
+++ crypto/bn/bn_convert.c
@@ -100,6 +100,9 @@ bn_bn2binpad_internal(const BIGNUM *bn, uint8_t *out,
 	if (out_len < n)
 		return -1;
 
+	if (out_len == 0)
+		return 0;
+
 	if (bn->dmax == 0) {
 		explicit_bzero(out, out_len);
 		return out_len;
--- tests/bn_convert.c.orig
+++ tests/bn_convert.c
@@ -62,7 +62,7 @@ check_bin_output(size_t test_no, const char *label, co
 		    "want %d\n", test_no, label, ret, out_len);
 		goto failure;
 	}
-	if (memcmp(out, bin, bin_len) != 0) {
+	if (out_len > 0 && memcmp(out, bin, bin_len) != 0) {
 		fprintf(stderr, "FAIL: Test %zu %s - output from "
 		    "BN_bn2bin() differs\n", test_no, label);
 		fprintf(stderr, "Got:\n");
