--- crypto/bn/bn_internal.h.orig	Sat Aug  8 21:05:00 2026
+++ crypto/bn/bn_internal.h	Sat Aug  8 21:05:00 2026
@@ -346,6 +346,12 @@ bn_subw_subw(BN_ULONG a, BN_ULONG b, BN_ULONG c, BN_ULONG *out_borrow,
  * 16 for 32 and 32 for 64.
  */
 #if 1
-static inline void
+#if defined(LIBRESSL_BN_MULW_NOINLINE)
+#define BN_MULW_INLINE __declspec(noinline)
+#else
+#define BN_MULW_INLINE inline
+#endif
+
+static BN_MULW_INLINE void
 bn_mulw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
 {
@@ -377,2 +383,5 @@ bn_mulw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
 }
+
+#undef BN_MULW_INLINE
+
 #else
