--- crypto/bn/bn_internal.h.orig
+++ crypto/bn/bn_internal.h
@@ -191,3 +191,8 @@
 #endif /* !BN_ULLONG */
-
+#if defined(LIBRESSL_BN_MONT_ARITHMETIC_NOINLINE)
+#define BN_MONT_ARITHMETIC_INLINE __declspec(noinline)
+#else
+#define BN_MONT_ARITHMETIC_INLINE inline
+#endif
+
 /*
@@ -198,3 +203,3 @@
 #ifndef HAVE_BN_ADDW
-static inline void
+static BN_MONT_ARITHMETIC_INLINE void
 bn_addw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
@@ -218,3 +223,3 @@
 #ifndef HAVE_BN_ADDW_ADDW
-static inline void
+static BN_MONT_ARITHMETIC_INLINE void
 bn_addw_addw(BN_ULONG a, BN_ULONG b, BN_ULONG c, BN_ULONG *out_r1,
@@ -348,3 +353,3 @@
 #if 1
-static inline void
+static BN_MONT_ARITHMETIC_INLINE void
 bn_mulw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
@@ -452,3 +457,3 @@
 #ifndef HAVE_BN_MULW_ADDW
-static inline void
+static BN_MONT_ARITHMETIC_INLINE void
 bn_mulw_addw(BN_ULONG a, BN_ULONG b, BN_ULONG c, BN_ULONG *out_r1,
@@ -472,3 +477,3 @@
 #ifndef HAVE_BN_MULW_ADDW_ADDW
-static inline void
+static BN_MONT_ARITHMETIC_INLINE void
 bn_mulw_addw_addw(BN_ULONG a, BN_ULONG b, BN_ULONG c, BN_ULONG d,
@@ -571,3 +576,3 @@
 #ifndef HAVE_BN_QWMULW_ADDQW_ADDW
-static inline void
+static BN_MONT_ARITHMETIC_INLINE void
 bn_qwmulw_addqw_addw(BN_ULONG a3, BN_ULONG a2, BN_ULONG a1, BN_ULONG a0,
@@ -592,2 +597,3 @@
-
+#undef BN_MONT_ARITHMETIC_INLINE
+
 #endif
