--- crypto/bn/arch/aarch64/bn_arch.h.orig
+++ crypto/bn/arch/aarch64/bn_arch.h
@@ -20,4 +20,27 @@
 #ifndef HEADER_BN_ARCH_H
 #define HEADER_BN_ARCH_H
-
+#if defined(_MSC_VER) && defined(_M_ARM64)
+
+#include <intrin.h>
+
+#define HAVE_BN_CT_NE_ZERO
+
+static inline int
+bn_ct_ne_zero(BN_ULONG w)
+{
+	return (w | (0 - w)) >> (BN_BITS2 - 1);
+}
+
+#define HAVE_BN_MULW
+
+static inline void
+bn_mulw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
+{
+	*out_r1 = (BN_ULONG)__umulh((unsigned __int64)a,
+	    (unsigned __int64)b);
+	*out_r0 = a * b;
+}
+
+#endif
+
 #ifndef OPENSSL_NO_ASM
