--- crypto/bn/bn_internal.h.orig	Sat Aug  8 21:50:00 2026
+++ crypto/bn/bn_internal.h	Sat Aug  8 21:50:00 2026
@@ -196,6 +196,12 @@ bn_mulw_addw_addw(BN_ULONG a, BN_ULONG b, BN_ULONG c, BN_ULONG d,
  * addition.
  */
 #ifndef HAVE_BN_ADDW
-static inline void
+#if defined(LIBRESSL_BN_ADDW_NOINLINE)
+#define BN_ADDW_INLINE __declspec(noinline)
+#else
+#define BN_ADDW_INLINE inline
+#endif
+
+static BN_ADDW_INLINE void
 bn_addw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
 {
@@ -211,2 +217,5 @@ bn_addw(BN_ULONG a, BN_ULONG b, BN_ULONG *out_r1, BN_ULONG *out_r0)
 }
+
+#undef BN_ADDW_INLINE
+
 #endif
